docs/charts/bar.md

```diff
@@ -112,6 +112,13 @@ Options are:
 * `'left'`
 * `'top'`
 * `'right'`
+* `false`
+
+#### borderWidth
+
+If this value is a number, it is applied to all sides of the rectangle (left, top, right, bottom), except [`borderSkipped`](#borderskipped). If this value is an object, the `left` property defines the left border width. Similarly the `right`, `top` and `bottom` properties can also be specified. Omitted borders are skipped.
+
+**Note:** for negative bars in vertical chart, `top` and `bottom` are flipped. Same goes for `left` and `right` in horizontal chart.
```

Comment:
This inversion comes from `boderSkipped` and has been that way some time. I don't think it should work differently for `borderWidth` in v2.

Reply:
That's a good point, the questions are:
- do we want to introduce a new feature which is initially inconsistent?
- do we want `borderSkipped` inverted and not `borderWidth`?

Reply:
I think they should both be inverted. When reading the note the first time it initially sounded confusing to me, but then I thought about it for a moment and thought that's the way I would want it to work. If I'm configuring this then as a user I would want it to consistently apply to the start or end of the bar. Maybe the names are what's confusing. Instead of skipping `top` and `bottom` maybe we should rename to `start` and `end` or something along those lines?

Reply:
What you are describing is a use case of `borderWidth`, specific to the bar controller that removes the border intersecting with the origin. In different contexts (e.g. floating bars), I would expect `top` to always be the top of the rectangle, whatever the value is fully positive or fully negative or in between.

Reply:
That's certainly the most common use case. I think I would still expect it to apply to `end` instead of `top` in the floating bar case as well. I tried finding example charts with the borders different on different sides and I can't find any. I'm sort of wondering why we're adding this feature of allowing configuring the border widths separately since no one has asked for it. I don't have any clear use cases in mind. All of the referenced tickets refer to `borderSkipped`. I thought the reason for adding this was so that you could set `borderWidth: { bottom: 0 }` instead of `borderSkipped: 'bottom'`. But if that's the case then `bottom` has to mean origin.

Reply:
I'm kind of in between. I think in `borderWidth`, `top` should mean `top`.
`borderSkipped` has to stay backward compatible, so `bottom` is origin there.

I think I would add 'start' and 'end' options to `borderSkipped` (and change defaults to those) if we want to keep that option available in v3. And I think it (or equivalent) should be available, because this use case for non-floating bar charts is common and should be easy to configure (without scripting).

Reply:
@etimberg @nagix thoughts?

Reply:
> I tried finding example charts with the borders different on different sides and I can't find any.
> I don't have any clear use cases in mind.

Let's say we also allow to configure the border color per side and someone wants green/thick for the max and red/thin for the min (e.g. floating bars):

```
borderSkipped: false,
borderWidth: {top: 4, bottom: 2},
borderColor: {top: 'green', bottom: 'red'}
```


Conversely, I can find only one use case where we want to invert vertical/horizontal edges: *make easier to remove the border at origin*. If that's the only reason why we want to keep the same inconsistent logic for `borderWidth`, I think we are going in the wrong direction (especially when introducing floating bars).

`borderSkipped` might have been introduced for the wrong reasons, thus the inverted logic. Adding support for `borderWidth` as object allows cases like #5071 but also some exotic designs where each edge can have a different width (and later color, etc.).

Reply:
Since this point impacts more the logic of this PR than the reviewed line, I posted a [new ticket comment](/chartjs/Chart.js/pull/6047#issuecomment-462924957) to continue the discussion at a more visible level.